File: currents_API_test.py
import requests

# API credentials
api_key = 'yujFLfzZuqpi8AenB-_aW_nMCNUx9r3XOq8t6YT_RlGb_u_d'

# API endpoint
url = 'https://api.currentsapi.services/v1/search'

# SQL-like query
query = "Elon AND (Space OR Tesla)"

# Request parameters
params = {
    'apiKey': api_key,
    'keywords': query,      # Complex query
    'language': 'en',
    'country': 'US',
    'page_size': 100,       # Fetch up to 100 results
    'start_date': '2025-02-14T00:00:00Z',
    'end_date': '2025-02-25T23:59:59Z',
}

# Make the API request
response = requests.get(url, params=params)

# Check the response
if response.status_code == 200:
    data = response.json()
    if data['status'] == 'ok':
        total_articles = len(data['news'])
        print(f"Total articles found: {total_articles}")
        for article in data['news']:
            print(article['title'])  # Print each article's title
    else:
        print("No results found or error in response.")
else:
    print(f"Error: {response.status_code}")

File: currents_API_test.js

const CurrentsAPI = require('currentsapi');
const currentsapi = new CurrentsAPI('yujFLfzZuqpi8AenB-_aW_nMCNUx9r3XOq8t6YT_RlGb_u_d');

// Set your start and end date
const startDate = '2025-01-01'; // Format: YYYY-MM-DD
const endDate = '2025-02-01';   // Format: YYYY-MM-DD

currentsapi.search({
  keywords: 'Trump',
  language: 'en',
  country: 'US',
  page_size: 100,
  from: startDate,    // Add the start date
  to: endDate         // Add the end date
}).then(response => {
  console.log(response);
  
  // Check if the response status is ok
  if (response.status === "ok") {
    console.log(`Total articles found: ${response.news.length}`);
  } else {
    console.log("No results found or error in response.");
  }
  /*
    Example output:
    {
      status: "ok",
      news: [...],
      totalArticles: 100
    }
  */
}).catch(error => {
  console.error("Error: ", error);
});
